src/main/java/com/ss/lms/controller/BorrowerController.java

```diff
@@ -78,4 +86,53 @@
 		return new ResponseEntity<Loans>(loan, HttpStatus.BAD_REQUEST);
 	}
 
+	@RequestMapping(path = "/api/branches", produces = { MediaType.APPLICATION_JSON_VALUE,
+			MediaType.APPLICATION_XML_VALUE })
+	public ResponseEntity<List<Branch>> getBranches() {
+		try {
+			List<Branch> branches = borrowerService.readBranch();
+			return new ResponseEntity<List<Branch>>(branches, HttpStatus.OK);
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			return new ResponseEntity<List<Branch>>(new ArrayList<Branch>(), HttpStatus.INTERNAL_SERVER_ERROR);
+
+		}
+
+	}
+	@RequestMapping(path = "/api/books", produces = { MediaType.APPLICATION_JSON_VALUE,
+			MediaType.APPLICATION_XML_VALUE })
+	public ResponseEntity<List<Book>> getBooks() {
+		try {
+			List<Book> books = borrowerService.readBooks();
+			return new ResponseEntity<List<Book>>(books, HttpStatus.OK);
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			return new ResponseEntity<List<Book>>(new ArrayList<Book>(), HttpStatus.INTERNAL_SERVER_ERROR);
+
+		}
+
+	}
+
+	@RequestMapping(path = "/api/books/{branchId}", produces = { MediaType.APPLICATION_JSON_VALUE,
```

Comment:
i think the correct path for this would be "/api/books/branches/{branchId}". Because in the version you have the branchId through typical API nomenclature would acutally be where the bookId would go.

Reply:
actually if your tryna get books by branchId believe the proper path would actually be "api/branches/{branchId}/books"

like for cds-> songs it would be "albums/{albumId}/songs" because the cd is a collection of the songs. The branch is essentially a collection of books